engine/node_matlab.cc

```diff
@@ -493,6 +493,20 @@ namespace MathDAG
     checkArg(0,0);
     return o<<"frac("<<arguments[0][0]->matlab()<<")";
   }
+  
+  template <>
+  ostream& OperationDAG<OperationType::percent>::matlab(ostream& o) const
+  {
+    checkArg(0,0);
+    return o<<"("<<arguments[0][0]->matlab()<<")%";
+  }  
+  
+  template <>
+  ostream& OperationDAG<OperationType::fact>::matlab(ostream& o) const
+  {
+    checkArg(0,0);
+    return o<<"factorial("<<arguments[0][0]->matlab()<<")";
```

Comment:
factorial is only defined for integers. I think you need to replace this with a call to gamma(1+arguments[0][0]->matlab())

Reply:
Did this.